File: app/src/main/java/com/ipd/taxiu/adapter/ExchangeRecordAdapter.java
package com.ipd.taxiu.adapter;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;

import com.ipd.taxiu.R;
import com.ipd.taxiu.ui.activity.coupon.CouponDetailActivity;
import com.ipd.taxiu.ui.activity.coupon.ExchangeRecordActivity;
import com.ipd.taxiu.ui.activity.coupon.ExchangeRecordDetailActivity;

import java.util.List;

/**
 * Created by Miss on 2018/7/20
 */
public class ExchangeRecordAdapter extends RecyclerView.Adapter<ExchangeRecordAdapter.ViewHolder> {

    private List<String> data;
    private Context mContext;

    public ExchangeRecordAdapter(List<String> data, Context mContext) {
        this.data = data;
        this.mContext = mContext;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            return new ViewHolder(View.inflate(mContext,R.layout.item_exchange_record,null));
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.rl_item.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(mContext, ExchangeRecordDetailActivity.class);
                mContext.startActivity(intent);
            }
        });
    }


    @Override
    public int getItemCount() {
        return data.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder{
        RelativeLayout rl_item;
        public ViewHolder(View itemView) {
            super(itemView);
            rl_item = itemView.findViewById(R.id.rl_exchange_record);
        }
    }
}

File: app/src/main/java/com/ipd/taxiu/adapter/ExchangeRecordAdapter.kt
package com.ipd.taxiu.adapter

import android.app.Activity
import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.ipd.taxiu.R
import com.ipd.taxiu.bean.ExchangeRecordBean
import com.ipd.taxiu.ui.activity.coupon.ExchangeRecordDetailActivity
import kotlinx.android.synthetic.main.item_exchange_record.view.*

/**
Created by Miss on 2018/8/13
 */
class ExchangeRecordAdapter(val context: Context, private val data: List<ExchangeRecordBean>) : RecyclerView.Adapter<ExchangeRecordAdapter.ViewHolder>(){
    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_exchange_record, parent, false))
    }

    override fun getItemCount(): Int = data?.size ?:0

    override fun onBindViewHolder(holder: ViewHolder?, position: Int) {
        holder?.itemView?.rl_exchange_record?.setOnClickListener { ExchangeRecordDetailActivity.launch(context as Activity) }
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}